### PR TITLE
[Algorithm] TD3+BC

### PR DESCRIPTION
**Pull request by BY571**
## Description

Adds the offline RL algorithm TD3+BC as described in the paper [A Minimalist Approach to
Offline Reinforcement Learning](https://arxiv.org/pdf/2106.06860).

Performance is verified for half-cheeta medium dataset:
image

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Documentation (update in the documentation)
- [X] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [X] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [X] My change requires a change to the documentation.
- [X] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: torchrl/objectives/td3_bc.py [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+        action_spec: TensorSpec = None,
+        bounds: Optional[Tuple[float]] = None,
+        num_qvalue_nets: int = 2,
+        policy_noise: float = 0.2,
+        noise_clip: float = 0.5,
+        alpha: float = 2.5,
+        loss_function: str = "smooth_l1",
+        delay_actor: bool = True,
+        delay_qvalue: bool = True,
+        gamma: float = None,
```
[the remark this answers is not in the document]
Comment (BY571) [reply torchrl/objectives/td3_bc.py:240]: Do you mean removing gamma?

**Review by vmoens — APPROVED** [review]
LGTM thanks!

File: torchrl/objectives/td3_bc.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vmoens) [reply torchrl/objectives/td3_bc.py:241]: yes we don't support gamma in constructors anymore

File: torchrl/objectives/td3_bc.py [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+from torchrl.objectives.value import TD0Estimator, TD1Estimator, TDLambdaEstimator
+
+
+class TD3BCLoss(LossModule):
+    r"""TD3+BC Loss Module.
+
+    Implementation of the TD3+BC loss presented in the paper `"A Minimalist Approach to
+    Offline Reinforcement Learning" <https://arxiv.org/pdf/2106.06860>`
+
+    Args:
```
Comment (vmoens) [inline torchrl/objectives/td3_bc.py:37]: Add the list of losses here along with the order in which they have to be executed.

File: torchrl/objectives/td3_bc.py [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+        return self.qvalue_network_params.detach()
+
+    @property
+    @_cache_values
+    def _cached_stack_actor_params(self):
+        return torch.stack(
+            [self.actor_network_params, self.target_actor_network_params], 0
+        )
+
+    def actor_loss(self, tensordict):
```
Comment (vmoens) [inline torchrl/objectives/td3_bc.py:372]: Add docstrings. See CrossQ for an example.

File: torchrl/objectives/td3_bc.py [hunk 4]
```diff
@@ -0,0 +1,536 @@
...
+
+        metadata = {
+            "state_action_value_actor": state_action_value_actor.detach(),
+            "bc_loss": bc_loss,
+            "lmbd": lmbd,
+        }
+        loss_actor = _reduce(loss_actor, reduction=self.reduction)
+        return loss_actor, metadata
+
+    def value_loss(self, tensordict):
```
Comment (vmoens) [inline torchrl/objectives/td3_bc.py:408]: Add docstrings. See CrossQ for an example.

File: torchrl/objectives/td3_bc.py [hunk 5]
```diff
@@ -0,0 +1,536 @@
...
+            "td_error": td_error,
+            "next_state_value": next_target_qvalue.detach(),
+            "pred_value": current_qvalue.detach(),
+            "target_value": target_value.detach(),
+        }
+        loss_qval = _reduce(loss_qval, reduction=self.reduction)
+        return loss_qval, metadata
+
+    @dispatch
+    def forward(self, tensordict: TensorDictBase) -> TensorDictBase:
```
Comment (vmoens) [inline torchrl/objectives/td3_bc.py:486]: Add docstrings. See CrossQ for an example.

File: test/test_cost.py [hunk 6]
```diff
@@ -2714,6 +2715,732 @@ def test_td3_reduction(self, reduction):
                 assert loss[key].shape == torch.Size([])
 
 
+@pytest.mark.skipif(
+    not _has_functorch, reason=f"functorch not installed: {FUNCTORCH_ERR}"
+)
```
Comment (vmoens) [inline test/test_cost.py:2720]:
```suggestion
```

**Review by vmoens — APPROVED** [review]
LGTM, just need a few more doctrings. I added some in CrossQ, feel free to take them as example

**Comment by BY571** [comment]
Docstrings are updated :)

File: torchrl/objectives/td3_bc.py [hunk 7]
```diff
@@ -398,14 +416,24 @@ def actor_loss(self, tensordict):
...
-    def value_loss(self, tensordict):
+    def qvalue_loss(self, tensordict):
+        """Compute the q-value loss.
+
+        The q-value loss should be computed before the :meth:`~.actor_loss`.
+
+        Args:
+            tensordict (TensorDictBase): the input data for the loss. Check the class's `in_keys` to see what fields
+                are required for this to be computed.
+        Returns: a differentiable tensor with the qvalue loss along with a metadata dictionary containing
```
Comment (vmoens) [inline torchrl/objectives/td3_bc.py:434]:
```suggestions
                are required for this to be computed.

        Returns: a differentiable tensor with the qvalue loss along with a metadata dictionary containing
```

**Pull request merged**